Engine/src/Engine/JobQueue.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Log.h"
+#include "JobQueue.h"
+
+using namespace engine_concurrent;
+
+JobQueue* JobQueue::instance = nullptr;
+std::mutex JobQueue::instanceMutex;
+std::mutex JobQueue::queueMutex;
+
+JobQueue* JobQueue::getInstance()
+{
+	std::unique_lock lock(instanceMutex);
+	if (instance == nullptr)
+	{
+		instance = new JobQueue();
+	}
+	return instance;
+}
+
+JobQueue::JobQueue() : m_jobList(new std::list<ThreadJob*>()) {}
+JobQueue::~JobQueue() { delete m_jobList; }

```

Comment:
since destructor is being declared here I would suggest deleting the move/copy/assign like the other classes in the PR